source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -123,7 +123,11 @@ namespace
                 continue;
             }
             const RtTypedValue* md = src->getMetadata(name);
-            dest->setAttribute(name.str(), md->getValueString());
+            std::string valueString = md->getValueString();
+            if (!valueString.empty())
+            {
+                dest->setAttribute(name.str(), md->getValueString());
```

Comment:
Why don't you use valueString above instead of retrieving it again?

Reply:
I was going to. Forgot to make the change :)